app/src/main/java/org/xwiki/android/sync/AppContext.kt

```diff
@@ -0,0 +1,120 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.android.sync
+
+import android.app.Application
+import android.util.Log
+import org.xwiki.android.sync.rest.BaseApiManager
+import org.xwiki.android.sync.utils.SharedPrefsUtils
+
+import java.util.AbstractMap
+import java.util.ArrayList
+
+/**
+ * Application class for authenticator
+ *
+ * @version $Id: c3a5996b1bce14d5c105a55f085115347c39c035 $
+ */
+class AppContext : Application() {
+
+    /**
+     * Set [.instance] to this object.
+     */
+    override fun onCreate() {
+        super.onCreate()
+        instance = this
+        Log.d(TAG, "on create")
+    }
+
+    companion object {
+
+        /**
+         * Entry pair Server address - Base Api Manager
+         */
+        private var baseApiManager: AbstractMap.SimpleEntry<String, BaseApiManager>? = null
+
+        /**
+         * Logging tag
+         */
+        private val TAG = "AppContext"
+
+        /**
+         * Instance of context to use it in static methods
+         */
+        /**
+         * @return known AppContext instance
+         */
+        var instance: AppContext? = null
```

Comment:
App context is always known why at least one component in android system is alive. I offer to replace it in file (instead of companion) and make it "lateinit var" with private setter

Reply:
But it was declared static and other static functions in this class are using it, IMO it should be here but I've added private and lateint property so it looks like this now: private lateinit var appContextInstance : AppContext